### PR TITLE
Update AWS SDK - WIP

### PR DESCRIPTION
**Pull request by szareiangm**
This is not complete yet. It will be finished tomorrow. I am adding @dmatth for any early feedback.

TODO:
- [x] Check previous blocking connections timeouts and add them here. All of them have 10 Seconds time-out: https://github.com/aws/aws-sdk-java/blob/55ac2d4a596b8c2ead9673057e7c07c38bdc8afa/aws-java-sdk-core/src/main/java/com/amazonaws/ClientConfiguration.java#L50 and https://aws.amazon.com/blogs/database/tuning-aws-java-sdk-http-request-settings-for-latency-aware-amazon-dynamodb-applications/
- [x] Check latest minor version of SDK
- [x] Check unit-tests

### CODE REVIEW

File: modules/kinesis/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/KinesisEnrich.scala [hunk 1]
```diff
@@ -42,8 +39,12 @@ import io.circe.Json
 import io.circe.syntax._
 import config._
 import model.{Credentials, DualCloudCredentialsPair, Kinesis, NoCredentials, SentryConfig, StreamsConfig}
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient
+import software.amazon.awssdk.services.dynamodb.model.{AttributeValue, GetItemRequest, ScanRequest}
 import sources.KinesisSource
-import utils.getAWSCredentialsProvider
+import utils.getAwsCredentialsProvider
```
Comment (szareiangm) [inline modules/kinesis/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/KinesisEnrich.scala:46]: removed the old credentials provider.

File: modules/kinesis/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/KinesisEnrich.scala [hunk 2]
```diff
@@ -71,7 +72,8 @@ object KinesisEnrich extends Enrich {
              forceDownload,
              credentials.aws,
              credentials.gcp,
-             Option(awsRegion)
+             Option(awsRegion),
+             new URI("http://todo-s3-endpo")
```
Comment (szareiangm) [inline modules/kinesis/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/KinesisEnrich.scala]: there are todo urls here and there, I have to propagate them in the code. due to the poor structure and tech-debt, it is not easy and I don't want put a new duct tape.

File: modules/kinesis/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/KinesisEnrich.scala [hunk 3]
```diff
@@ -168,26 +170,39 @@ object KinesisEnrich extends Enrich {
...
+                      .build()
+                  )
+                  .get(10, TimeUnit.SECONDS)
+                  .item()
+              ).fold(s"Key $key doesn't exist in DynamoDB table $table".asLeft[Map[String, AttributeValue]])(
+                _.asScala.toMap.asRight[String]
+              )
+      json <- item
+                .get("json")
+                .fold(s"""Field "json" not found at key $key in DynamoDB table $table""".asLeft[String])(_.s().asRight[String])
```
Comment (szareiangm) [inline modules/kinesis/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/KinesisEnrich.scala:199]: I really liked `fold`. a good lesson for us to use to validate options in a fancy inline way.

File: modules/kinesis/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/sinks/KinesisSink.scala [hunk 4]
```diff
@@ -23,37 +23,29 @@ package sinks
...
 object KinesisSink {
-  def validate(kinesisConfig: Kinesis, streamName: String): Either[String, Unit] =
+  def validate(client: KinesisAsyncClient, streamName: String): Either[String, Unit] =
     for {
-      provider <- getAWSCredentialsProvider(kinesisConfig.aws)
-      endpointConfiguration = new EndpointConfiguration(
-                                kinesisConfig.streamEndpoint,
-                                kinesisConfig.region
-                              )
-      client = AmazonKinesisClientBuilder
```
Comment (szareiangm) [inline modules/kinesis/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/sinks/KinesisSink.scala:52]: this was too much. creating excessive client. I just passed it from top layer.

**Review by dmatth — COMMENTED** [review]
looks OK

**Pull request merged**